### PR TITLE
Upgrade TiDB to 4.0.0-beta, and recognize @@tidb_row_format_version = 2

### LINKED ISSUE #267: *: use new row-format in tidb
## Feature Request

**Is your feature request related to a problem? Please describe:**

TiDB v4.0 uses new row format https://github.com/pingcap/tidb/blob/master/docs/design/2018-07-19-row-format.md format, lightning should also update to use new row format

**Describe alternatives you've considered:**

It is best to use the tidb library functions to construct row data directly, so that you only need to update the tidb version in the future.

### PR DESCRIPTION
**Pull request by kennytm**
### What problem does this PR solve? 

1. Regular dependency upgrade, mainly TiDB → v4.0.0-beta.
2. Fix #267 by copying the value of `@@tidb_row_format_version` from downstream if exists.

### What is changed and how it works?

↑

### Check List 

Tests 

 - Unit test
 - Integration test

Side effects

Related changes

### CODE REVIEW

File: lightning/backend/sql2kv.go [hunk 1]
```diff
@@ -124,19 +124,19 @@ func logKVConvertFailed(logger log.Logger, row []types.Datum, j int, colInfo *mo
 	)
 }
 
-type kvPairs []kvec.KvPair
```
Comment (GregoryIan) [inline lightning/backend/sql2kv.go]: Did `tidb` remove the  `kvencoder` pkg?

File: lightning/backend/allocator.go [hunk 2]
```diff
@@ -19,33 +19,43 @@ import (
 	"github.com/pingcap/tidb/meta/autoid"
 )
 
-// PanickingAllocator is an ID allocator which panics on all operations except Rebase
-type PanickingAllocator struct {
+// panickingAllocator is an ID allocator which panics on all operations except Rebase
+type panickingAllocator struct {
 	autoid.Allocator
-	base int64
+	base *int64
```
Comment (siddontang) [inline lightning/backend/allocator.go]: em, why change to a pointer?

File: lightning/backend/sql2kv.go (hunk 1 above)
Comment (kennytm) [reply lightning/backend/sql2kv.go]: Yes it did

File: lightning/backend/allocator.go (hunk 2 above)
Comment (kennytm) [reply lightning/backend/allocator.go]: So that all 3 allocators (row_id, auto_inc, auto_random) share the same value when we Rebase

**Comment by kennytm** [comment]
/run-all-tests tidb=release-3.0 tikv=release-3.0 pd=release-3.0

**Comment by kennytm** [comment]
/run-all-tests tidb=release-3.0 tikv=release-3.0 pd=release-3.0

**Comment by kennytm** [comment]
PTAL @3pointer cc @GregoryIan

File: tests/row-format-v2/run.sh [hunk 3]
```diff
@@ -0,0 +1,21 @@
+#!/bin/sh
+
+# Basic check for correctness when row format v2 is active.
+# (There's no way to verify if the rows are really in format v2 though...)
```
Comment (GregoryIan) [inline tests/row-format-v2/run.sh]: 😷

**Comment by GregoryIan** [comment]
LGTM

**Pull request merged**